### PR TITLE
docs(ReactionCollector): Revise JSDoc for ReactionCollector#dispose and #remove

### PR DESCRIPTION
**Pull request by qwright10**
**Please describe the changes this PR makes and why it should be merged:**

**Status**

- [x] Code changes have been tested against the Discord API, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**

- [ ] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [x] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/structures/ReactionCollector.js [hunk 1]
```diff
@@ -103,17 +103,15 @@ class ReactionCollector extends Collector {
    */
   dispose(reaction, user) {
     /**
-     * Emitted whenever a reaction is disposed of and the `dispose` option is set to true.
+     * Emitted when the entire reaction is removed and the `dispose` option is set to true.
```
Comment (kyranet) [inline src/structures/ReactionCollector.js:106]:
```suggestion
     * Emitted when the reaction had all the users removed and the `dispose` option is set to true.
```

File: src/structures/ReactionCollector.js [hunk 2]
```diff
@@ -103,17 +103,15 @@ class ReactionCollector extends Collector {
...
      * @param {MessageReaction} reaction The reaction that was disposed of
      * @param {User} user The user that removed the reaction
      */
     if (reaction.message.id !== this.message.id) return null;
 
     /**
-     * Emitted whenever a reaction is removed from a message and the `dispose` option is set to true.
-     * Will emit on all reaction removals, as opposed to {@link Collector#dispose} which will only
-     * be emitted when the entire reaction is removed.
+     * Emitted when a single reaction is removed and the `dispose` option is set to true.
```
Comment (kyranet) [inline src/structures/ReactionCollector.js:114]:
```suggestion
     * Emitted when the reaction had one user removed and the `dispose` option is set to true.
```

**Review by kyranet — CHANGES_REQUESTED** [review]
Alternative wording, "single reaction" and "entire reaction" means the same, as both refer to "one reaction", however, we are talking about users being removed here, not the reactions themselves.

File: src/structures/ReactionCollector.js (hunk 1 above)
Comment (Papaia) [reply src/structures/ReactionCollector.js:106]: I tend to disagree, the wording should be changed but "removing a user" just doesn't feel right, I don't remove myself from the reaction, I remove my reaction, and a user removes their reaction, and when all reactions are removed at once it is very much like that.

Comment (tipakA) [reply src/structures/ReactionCollector.js:106]: No. You remove yourself from users who reacted.
Message has reactions. Each reaction has an emoji. Each reaction has collection of users who reacted to it. If you un-react, you don't delete reaction at all. Reaction happens to be removed if there are 0 users who reacted with that emoji.

**Comment by almostSouji** [comment]
> Alternative wording, "single reaction" and "entire reaction" means the same, as both refer to "one reaction", however, we are talking about users being removed here, not the reactions themselves.

In my humble opinion (and how i tend to explain it in support so people understand it and keep consistency with the bot methods that make these events emits)
- A user is removed from a reaction (MessageReaction#users#remove thought train)
- A reaction is removed (MessageReaction#remove)
- All reactions are removed from a message (ReactionManager#removeAll)

**Review by iCrawl — APPROVED** [review]
We can always improve this whenever someone comes up with better wording, but for now this LGTM

**Review by vladfrangu — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Pull request merged**